java-checks/src/main/java/org/sonar/java/filters/LombokFilter.java

```diff
@@ -93,56 +99,44 @@
   }
 
   @Override
-  public void visitClass(ClassTree tree) {
-    boolean generatesEquals = usesAnnotation(tree, GENERATE_EQUALS);
+  public void visitImport(ImportTree tree) {
+    String fullyQualifiedName = ExpressionsHelper.concatenate((ExpressionTree) tree.qualifiedIdentifier());
 
-    if (generatesEquals || usesAnnotation(tree, GENERATE_UNUSED_FIELD_RELATED_METHODS)) {
-      excludeLines(tree, UnusedPrivateFieldCheck.class);
-      excludeLines(tree, PrivateFieldUsedLocallyCheck.class);
-    } else {
-      acceptLines(tree, UnusedPrivateFieldCheck.class);
-      acceptLines(tree, PrivateFieldUsedLocallyCheck.class);
-    }
+    excludeLinesIfTrue(fullyQualifiedName.startsWith("lombok."), tree, UselessImportCheck.class);
 
-    if (usesAnnotation(tree, GENERATE_CONSTRUCTOR)) {
-      excludeLines(tree, AtLeastOneConstructorCheck.class);
-    } else {
-      acceptLines(tree, AtLeastOneConstructorCheck.class);
-    }
+    super.visitImport(tree);
+  }
 
-    if (generatesEquals) {
-      excludeLines(tree, EqualsNotOverriddenInSubclassCheck.class);
-      excludeLines(tree, EqualsNotOverridenWithCompareToCheck.class);
-    } else {
-      acceptLines(tree, EqualsNotOverriddenInSubclassCheck.class);
-      acceptLines(tree, EqualsNotOverridenWithCompareToCheck.class);
-    }
+  @Override
+  public void visitClass(ClassTree tree) {
+    boolean generatesEquals = usesAnnotation(tree, GENERATE_EQUALS);
 
-    if (generatesPrivateConstructor(tree)) {
-      excludeLines(tree, UtilityClassWithPublicConstructorCheck.class);
-    } else {
-      acceptLines(tree, UtilityClassWithPublicConstructorCheck.class);
-    }
+    excludeLinesIfTrue(generatesEquals || usesAnnotation(tree, GENERATE_UNUSED_FIELD_RELATED_METHODS), tree, UnusedPrivateFieldCheck.class, PrivateFieldUsedLocallyCheck.class);
+    excludeLinesIfTrue(usesAnnotation(tree, GENERATE_CONSTRUCTOR), tree, AtLeastOneConstructorCheck.class);
+    excludeLinesIfTrue(generatesEquals, tree, EqualsNotOverriddenInSubclassCheck.class, EqualsNotOverridenWithCompareToCheck.class);
+    excludeLinesIfTrue(generatesPrivateConstructor(tree), tree, UtilityClassWithPublicConstructorCheck.class);
+    excludeLinesIfTrue(usesAnnotation(tree, UTILITY_CLASS), tree, BadFieldNameCheck.class, ConstantsShouldBeStaticFinalCheck.class);
+
+    super.visitClass(tree);
+  }
 
-    if (usesAnnotation(tree, UTILITY_CLASS)) {
-      excludeLines(tree, BadFieldNameCheck.class);
-      excludeLines(tree, ConstantsShouldBeStaticFinalCheck.class);
+  private final void excludeLinesIfTrue(boolean shouldExclude, Tree tree, Class<? extends JavaCheck> rule) {
```

Comment:
`final` doesn't make sense for private method, we should have a rule for that

Reply:
this one is indeed a mistake. the `final` was forced for the other one because of the `@SafeVarargs` annotation